quipucords/scanner/openshift/api.py

```diff
@@ -106,18 +106,35 @@ def _dynamic_client(self):
 
     @classmethod
     def with_config_info(
-        cls, *, host, protocol, port, ssl_verify: bool = True, **kwargs
+        cls,
+        *,
+        host,
+        protocol,
+        port,
+        ssl_verify: bool = True,
+        http_proxy: str = None,
+        https_proxy: str = None,
+        **kwargs,
     ):
         """Initialize OpenShiftApi without providing a KubeConfig object."""
         host_uri = f"{protocol}://{host}:{port}"
+        proxy_url = None
+        if http_proxy:
+            proxy_url = "http://" + http_proxy
+        elif https_proxy:
+            proxy_url = "http://" + https_proxy
```

Comment:
aaaaaaaaand this is deeply related to my question here: https://github.com/quipucords/quipucords/pull/2863/files#r2060281078

this gets worse at this point, as a incorrect proxy config can cause a override and the user will scratch their head trying to figure out why their proxy config is not working.

With this in mind, I'd change the proxy related column on `Source` model and simply call it `proxy_url` (taking inspiration on the nomenclature used here). As a validation, I'd check if the protocol is the same used on the source host url.